### PR TITLE
Simplify some code surrounding the use of SolutionTransfer.

### PR DESCRIPTION
**Pull request by bangerth**
C++11 allows to simplify things a bit.

### CODE REVIEW

**Review by gassmoeller — APPROVED** [review]
Looks nicer this way too. Thanks.

**Comment by gassmoeller** [comment]
But you will have to check what happens with the tests.

File: source/simulator/core.cc [hunk 1]
```diff
@@ -1711,6 +1713,12 @@ namespace aspect
     {
       TimerOutput::Scope timer (computing_timer, "Refine mesh structure, part 2");
 
+      parallel::distributed::SolutionTransfer<dim,LinearAlgebra::BlockVector>
+      system_trans(dof_handler);
```
Comment (tjhei) [inline source/simulator/core.cc:1717]: I think this local variable needs to be deleted as it shadows the one above.

Comment (bangerth) [reply source/simulator/core.cc]: Yes, exactly. I put the declarations into the wrong scope. Let's see what happens now.

**Comment by bangerth** [comment]
OK, works now!

**Review by gassmoeller — APPROVED** [review]

**Pull request merged**